src/pudl/analysis/ferc1_eia.py

```diff
@@ -431,596 +436,139 @@ def get_features(self, clobber=False):
-        matches_best_df.loc[
-            fill_in_the_blank_rules, "match_type"
-        ] = "no prediction; training"
+def run_model(features_train, features_all, train_df):
+    """Train Linear Regression model using GridSearch.
 
-        logger.info(
-            f"Overridden records:       {len(overridden)/len(train_df):.01%}\n"
-            "New best match v ferc:    "
-            f"{len(matches_best_df)/self.ferc1_options_len:.02%}"
-        )
-        # we don't need these cols anymore...
-        matches_best_df = matches_best_df.drop(
-            columns=["record_id_eia_trn", "record_id_eia_rl", "record_id_ferc1_trn"]
-        )
-        return matches_best_df
-
-    @staticmethod
-    def fit_predict_lrc(best, features_train, features_all, train_df):
-        """Generate, fit and predict model.
-
-        Wahoo.
-        """
-        # prep the model with the hyperparameters
-        lrc = rl.LogisticRegressionClassifier(
-            solver=best["solver"].values[0],
-            C=best["c"].values[0],
-            class_weight=best["cw"].values[0],
-            penalty=best["penalty"].values[0],
-            l1_ratio=None if pd.isnull(best["l1"].values[0]) else best["l1"].values[0],
-            random_state=0,
-            multi_class=best["multi_class"].values[0],
-        )
-        # fit the model with all of the
-        lrc.fit(features_train, train_df.index)
-        # this step is getting preditions on all of the possible matches based
-        # on the last run model above
-        predict_all = lrc.predict(features_all)
-        predict_all_df = pd.merge(
-            pd.DataFrame(index=predict_all),
-            features_all,
+    Fit model for best parameters and make predictions of matches on all input features.
+    """
+    param_grid = [
+        {
+            "solver": ["newton-cg", "lbfgs", "sag"],
+            "C": [1000, 1, 10, 100],
+            "class_weight": [None, "balanced"],
+            "penalty": ["l2"],
+        },
+        {
+            "solver": ["liblinear", "saga"],
+            "C": [1000, 1, 10, 100],
+            "class_weight": [None, "balanced"],
+            "penalty": ["l1", "l2"],
+        },
+        {
+            "solver": ["saga"],
+            "C": [1000, 1, 10, 100],
+            "class_weight": [None, "balanced"],
+            "penalty": ["elasticnet"],
+            "l1_ratio": [0.1, 0.3, 0.5, 0.7, 0.9],
+        },
+    ]
+    x_train = features_train.to_numpy()
+    y_train = np.where(
+        features_train.merge(
+            train_df,
+            how="left",
             left_index=True,
             right_index=True,
-            how="left",
-        )
-        return predict_all_df
+            indicator=True,
+        )["_merge"]
+        == "both",
+        1,
+        0,
+    )
+    lrc = LogisticRegression()
+    clf = GridSearchCV(estimator=lrc, param_grid=param_grid, verbose=True, n_jobs=-1)
```

Comment:
`GridSearchCV` performs K-fold cross validation on every combination of parameters from `param_grid` and returns the best performing model. It is parallelized (`n_jobs=-1`) which is why it's so much faster.